config_example.yaml

```diff
@@ -8,7 +8,13 @@ customfields: # custom fields are added to falco events and metrics, if the valu
 templatedfields: # templated fields are added to falco events and metrics, it uses Go template + output_fields values
   # Dkey: '{{ or (index . "k8s.ns.labels.foo") "bar" }}'
 # bracketreplacer: "_" # if not empty, the brackets in keys of Output Fields are replaced
-mutualtlsfilespath: "/etc/certs" # folder which will used to store client.crt, client.key and ca.crt files for mutual tls (default: "/etc/certs")
+mutualtlsfilespath: "/etc/certs" # folder which will used to store client.crt, client.key and ca.crt files for mutual tls for outputs (default: "/etc/certs")
+tlsserver:
+  deploy: false # if true, TLS server will be deployed instead of HTTP
+  certfile: "/etc/certs/server.crt" # server certification file
+  keyfile: "/etc/certs/server.key" # server key
+  mutualtls: false # if true, mTLS server will be deployed instead of TLS, deploy also has to be true
+  cacertfile: "/etc/certs/ca.crt" # for client certification if mutualtls is true
```

Comment:
I'm afraid it will create confusion for end users. We have 2 different mTLS:
- the mTLS used to authenticate falcosidekick with its outputs
- the mTLS used by the falcosidekick's webserver
So we can't use the same folder as they might have conflicts with the `ca.crt `, wdyt?

Reply:
I agree it could be confusing that the default path is the same folder. They could rename the `ca.crt` for the server to something else if they need two different ones though.
One solution I see is that we could use `etc/certs/server` for the new ones and `/etc/certs/client` for the already existing ones - this sounds more logical in my mind but this would mean breaking change for those who used the default option.
Other path name that I would suggest is `/etc/tls/server` and leave the client one as is. Just to have something different and in the end everyone can set it to whatever they want.

Reply:
Just to be clear, you mean something like this?
```
mutualtlsfilespath: "/etc/certs" # folder which will used to store client.crt, client.key and ca.crt files for mutual tls for outputs, will be deprecated in the future (default: "/etc/certs")
mutualtlsclient:
  certfile: "/etc/certs/client/client.crt" # client certification file
  keyfile: "/etc/certs/client/client.key" # client key
  cacertfile: "/etc/certs/client/ca.crt" # for server certification
tlsserver:
  deploy: false # if true, TLS server will be deployed instead of HTTP
  certfile: "/etc/certs/server/server.crt" # server certification file
  keyfile: "/etc/certs/server/server.key" # server key
  mutualtls: false # if true, mTLS server will be deployed instead of TLS, deploy also has to be true
  cacertfile: "/etc/certs/server/ca.crt" # for client certification if mutualtls is true
```

Reply:
Ok with your solution.

For the `mutualtls`, do you think we could enforce deploy if it's true?

Reply:
Great!

I think that would be confusing eg. if I turn off the `deploy` for debugging, then I would expect that the whole TLS part is turned off, but it would still be on. Maybe I could set a warning log message for the `deploy == false && mutualtls == true` case.

Reply:
Ok for the warning, thanks

Reply:
Added the modifications in a new commit, let me know if it's okay with you or I should change something!